test/unit/utils/test-xhr-utils.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as mode from '../../../src/mode';
+import * as url from '../../../src/url';
+import {
+  ALLOW_SOURCE_ORIGIN_HEADER,
+  getViewerInterceptResponse,
+  setupAMPCors,
+  setupInit,
+  verifyAmpCORSHeaders,
+} from '../../../src/utils/xhr-utils';
+import {Services} from '../../../src/services';
+
+describes.sandboxed('utils/xhr-utils', {}, env => {
+
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = env.sandbox;
+  });
+
+  describe('verifyAmpCORSHeaders', () => {
+    it('should verify allowed source origin', () => {
+      const sourceOrigin = 'https://www.da-original.org';
+      const headers = {};
+      headers[ALLOW_SOURCE_ORIGIN_HEADER] = sourceOrigin;
+      const response = new Response({}, {headers: new Headers(headers)});
+      const win = {
+        location: {
+          href: sourceOrigin,
+        },
+      };
+      expect(() => {
+        verifyAmpCORSHeaders(win, response, {} /* init */);}).to.not.throw;
+    });
+
+    it('should throw error if invalid origin', () => {
+      const sourceOrigin = 'https://www.da-original.org';
+      const headers = {};
+      headers[ALLOW_SOURCE_ORIGIN_HEADER] = 'https://www.original.org';
+      const response = new Response({}, {headers: new Headers(headers)});
+      const win = {
+        location: {
+          href: sourceOrigin,
+        },
+      };
+      allowConsoleError(() => {
```

Comment:
Remove `allowConsoleError`. You actually need to return it, otherwise the embedded code won't execute before the test completes.